### ISSUE TITLE
Script kill's when disconnected

### ISSUE THREAD

**Issue by arthurgeek** [slackhq/hubot-slack#224]
Hi,

We've been using this for a while now. A month ago we were using the 2.2 version, and everything was fine. But one day, our client got disabled, and we couldn't reenable it and keep using the old version, We had to update to a new version (API changes in Slack). So, we upgraded to 3.3 version, and a lot of weird things started to happen. Our hubot client runs on docker, so, every time the main hub process ends, so does the docker container. This wasn't happening before. Yesterday I started using the master branch version, as it looks this issue was solved, but we still have the same problem. How to solve this problem? I want to keep using this as it ever worked before. The disconnection issue shouldn't bring the client down, it doesn't makes any sense.

Here's the latest log:

```
[Wed Sep 02 2015 17:32:46 GMT+0000 (UTC)] ERROR Last pong is too old: 10.004
[Wed Sep 02 2015 17:32:46 GMT+0000 (UTC)] INFO Reconnecting in 1000ms
[Wed Sep 02 2015 17:32:47 GMT+0000 (UTC)] INFO Attempting reconnect
[Wed Sep 02 2015 17:32:47 GMT+0000 (UTC)] INFO Connecting...
[Wed Sep 02 2015 17:32:53 GMT+0000 (UTC)] ERROR Received error "failed to read response string"
[Wed Sep 02 2015 17:32:53 GMT+0000 (UTC)] ERROR undefined
[Wed Sep 02 2015 17:32:53 GMT+0000 (UTC)] ERROR Exiting in 1 second
[Wed Sep 02 2015 17:32:53 GMT+0000 (UTC)] INFO Reconnecting in 2000ms
```

And then the client just quit.

**Comment by technicalpickles** [comment]
I'm wondering, do any of the versions between 2.2 and 3.3 work for you? There's 3.0.0, 3.1.0, 3.2.0, and 3.2.1. That might help narrow down if this is a new thing or not.

I don't see anything else obvious in the output. Would you mind including the output of `node --version`, `npm --version`, and `npm ls`?

**Comment by technicalpickles** [comment]
That's interesting, a ruby bot has that same "failed to read response string", so I suspect the message is originating from the Slack API: https://github.com/dblock/slack-ruby-bot/issues/15

**Comment by arthurgeek** [comment]
@technicalpickles thanks for the info, I'll try with older versions and see if the problem persists.

The outputs you asked:

```
hubot@8a0a47356723:/app# node --version
v0.10.25
hubot@8a0a47356723:/app# npm --version
1.3.10
hubot@8a0a47356723:/app# npm ls
hosted-hubot@2.7.1 /app
├── ascii-table@0.0.8
├─┬ aws-sdk@2.1.49
│ ├── sax@0.5.3
│ └── xmlbuilder@0.4.2
├─┬ aws2js@0.6.22
│ ├─┬ libxml-to-js@0.3.11
│ │ └── lodash@3.10.1
│ ├─┬ libxmljs@0.14.3
│ │ ├── bindings@1.2.1
│ │ └── nan@2.0.7
│ └─┬ mime-magic@5.14.0
│   └── config@1.15.0
├─┬ cheerio@0.12.4
│ ├─┬ cheerio-select@0.0.3
│ │ └─┬ CSSselect@0.7.0
│ │   ├── boolbase@1.0.0
│ │   ├── CSSwhat@0.4.7
│ │   ├─┬ domutils@1.4.3
│ │   │ └── domelementtype@1.3.0
│ │   └── nth-check@1.0.1
│ ├── entities@0.5.0
│ ├─┬ htmlparser2@3.1.4
│ │ ├── domelementtype@1.3.0
│ │ ├── domhandler@2.0.3
│ │ ├── domutils@1.1.6
│ │ └─┬ readable-stream@1.0.33
│ │   ├── core-util-is@1.0.1
│ │   ├── inherits@2.0.1
│ │   ├── isarray@0.0.1
│ │   └── string_decoder@0.10.31
│ └── underscore@1.4.4
├── date-utils@1.2.17
├─┬ githubot@0.4.2
│ ├── async@0.2.10
│ └── scoped-http-client@0.11.0
├── htmlparser@1.7.7
├─┬ hubot@2.16.0
│ ├── async@0.9.2
│ ├─┬ chalk@1.1.1
│ │ ├── ansi-styles@2.1.0
│ │ ├── escape-string-regexp@1.0.3
│ │ ├─┬ has-ansi@2.0.0
│ │ │ └── ansi-regex@2.0.0
│ │ ├─┬ strip-ansi@3.0.0
│ │ │ └── ansi-regex@2.0.0
│ │ └── supports-color@2.0.0
│ ├── cline@0.8.2
│ ├── coffee-script@1.6.3
│ ├─┬ connect-multiparty@1.2.5
│ │ ├─┬ multiparty@3.3.2
│ │ │ ├─┬ readable-stream@1.1.13
│ │ │ │ ├── core-util-is@1.0.1
│ │ │ │ ├── inherits@2.0.1
│ │ │ │ ├── isarray@0.0.1
│ │ │ │ └── string_decoder@0.10.31
│ │ │ └── stream-counter@0.2.0
│ │ ├─┬ on-finished@2.1.1
│ │ │ └── ee-first@1.1.0
│ │ ├── qs@2.2.5
│ │ └─┬ type-is@1.5.7
│ │   ├── media-typer@0.3.0
│ │   └─┬ mime-types@2.0.14
│ │     └── mime-db@1.12.0
│ ├─┬ express@3.18.1
│ │ ├── basic-auth@1.0.0
│ │ ├─┬ commander@1.3.2
│ │ │ └── keypress@0.1.0
│ │ ├─┬ connect@2.27.1
│ │ │ ├── basic-auth-connect@1.0.0
│ │ │ ├─┬ body-parser@1.9.3
│ │ │ │ ├── iconv-lite@0.4.5
│ │ │ │ ├─┬ on-finished@2.1.1
│ │ │ │ │ └── ee-first@1.1.0
│ │ │ │ ├── qs@2.3.3
│ │ │ │ └── raw-body@1.3.1
│ │ │ ├── bytes@1.0.0
│ │ │ ├─┬ compression@1.2.2
│ │ │ │ ├─┬ accepts@1.1.4
│ │ │ │ │ ├─┬ mime-types@2.0.14
│ │ │ │ │ │ └── mime-db@1.12.0
│ │ │ │ │ └── negotiator@0.4.9
│ │ │ │ └─┬ compressible@2.0.5
│ │ │ │   └── mime-db@1.17.0
│ │ │ ├─┬ connect-timeout@1.4.0
│ │ │ │ └── ms@0.6.2
│ │ │ ├─┬ cookie-parser@1.3.5
│ │ │ │ ├── cookie@0.1.3
│ │ │ │ └── cookie-signature@1.0.6
│ │ │ ├─┬ csurf@1.6.6
│ │ │ │ └─┬ csrf@2.0.7
│ │ │ │   ├── base64-url@1.2.1
│ │ │ │   ├── rndm@1.1.0
│ │ │ │   ├── scmp@1.0.0
│ │ │ │   └─┬ uid-safe@1.1.0
│ │ │ │     └── native-or-bluebird@1.1.2
│ │ │ ├─┬ errorhandler@1.2.4
│ │ │ │ └─┬ accepts@1.1.4
│ │ │ │   ├─┬ mime-types@2.0.14
│ │ │ │   │ └── mime-db@1.12.0
│ │ │ │   └── negotiator@0.4.9
│ │ │ ├─┬ express-session@1.9.3
│ │ │ │ ├── crc@3.2.1
│ │ │ │ └─┬ uid-safe@1.0.1
│ │ │ │   ├── base64-url@1.2.1
│ │ │ │   └─┬ mz@1.3.0
│ │ │ │     ├── native-or-bluebird@1.2.0
│ │ │ │     ├── thenify@3.1.0
│ │ │ │     └── thenify-all@1.6.0
│ │ │ ├─┬ finalhandler@0.3.2
│ │ │ │ └─┬ on-finished@2.1.1
│ │ │ │   └── ee-first@1.1.0
│ │ │ ├─┬ http-errors@1.2.8
│ │ │ │ ├── inherits@2.0.1
│ │ │ │ └── statuses@1.2.1
│ │ │ ├─┬ method-override@2.3.5
│ │ │ │ ├─┬ debug@2.2.0
│ │ │ │ │ └── ms@0.7.1
│ │ │ │ └── methods@1.1.1
│ │ │ ├─┬ morgan@1.4.1
│ │ │ │ └─┬ on-finished@2.1.1
│ │ │ │   └── ee-first@1.1.0
│ │ │ ├─┬ multiparty@3.3.2
│ │ │ │ ├─┬ readable-stream@1.1.13
│ │ │ │ │ ├── core-util-is@1.0.1
│ │ │ │ │ ├── inherits@2.0.1
│ │ │ │ │ ├── isarray@0.0.1
│ │ │ │ │ └── string_decoder@0.10.31
│ │ │ │ └── stream-counter@0.2.0
│ │ │ ├── on-headers@1.0.0
│ │ │ ├── pause@0.0.1
│ │ │ ├── qs@2.3.0
│ │ │ ├── response-time@2.2.0
│ │ │ ├─┬ serve-favicon@2.1.7
│ │ │ │ └── ms@0.6.2
│ │ │ ├─┬ serve-index@1.5.3
│ │ │ │ ├─┬ accepts@1.1.4
│ │ │ │ │ └── negotiator@0.4.9
│ │ │ │ ├── batch@0.5.1
│ │ │ │ └─┬ mime-types@2.0.14
│ │ │ │   └── mime-db@1.12.0
│ │ │ ├── serve-static@1.7.2
│ │ │ ├─┬ type-is@1.5.7
│ │ │ │ └─┬ mime-types@2.0.14
│ │ │ │   └── mime-db@1.12.0
│ │ │ └── vhost@3.0.1
│ │ ├── content-disposition@0.5.0
│ │ ├── cookie@0.1.2
│ │ ├── cookie-signature@1.0.5
│ │ ├─┬ debug@2.1.3
│ │ │ └── ms@0.7.0
│ │ ├── depd@1.0.1
│ │ ├── escape-html@1.0.1
│ │ ├─┬ etag@1.5.1
│ │ │ └── crc@3.2.1
│ │ ├── fresh@0.2.4
│ │ ├── media-typer@0.3.0
│ │ ├── merge-descriptors@0.0.2
│ │ ├── methods@1.1.0
│ │ ├─┬ mkdirp@0.5.0
│ │ │ └── minimist@0.0.8
│ │ ├── parseurl@1.3.0
│ │ ├─┬ proxy-addr@1.0.8
│ │ │ ├── forwarded@0.1.0
│ │ │ └── ipaddr.js@1.0.1
│ │ ├── range-parser@1.0.2
│ │ ├─┬ send@0.10.1
│ │ │ ├── destroy@1.0.3
│ │ │ ├── mime@1.2.11
│ │ │ ├── ms@0.6.2
│ │ │ └─┬ on-finished@2.1.1
│ │ │   └── ee-first@1.1.0
│ │ ├── utils-merge@1.0.0
│ │ └── vary@1.0.1
│ ├── log@1.4.0
│ ├── optparse@1.0.4
│ └── scoped-http-client@0.11.0
├─┬ hubot-scripts@2.5.16
│ └── redis@0.8.4
├─┬ hubot-slack@3.3.0 (git://github.com/slackhq/hubot-slack#af37d933671423e5c00966395c9852ee7123bab7)
│ └─┬ slack-client@1.4.1
│   ├── coffee-script@1.9.3
│   ├── log@1.4.0
│   └─┬ ws@0.4.31
│     ├── commander@0.6.1
│     ├── nan@0.3.2
│     ├── options@0.0.6
│     └── tinycolor@0.0.1
├─┬ jsdom@1.0.3
│ ├── browser-request@0.3.3
│ ├─┬ contextify@0.1.14
│ │ ├── bindings@1.2.1
│ │ └── nan@1.8.4
│ ├── cssom@0.3.0
│ ├── cssstyle@0.2.29
│ ├─┬ htmlparser2@3.8.3
│ │ ├── domelementtype@1.3.0
│ │ ├── domhandler@2.3.0
│ │ ├─┬ domutils@1.5.1
│ │ │ └─┬ dom-serializer@0.1.0
│ │ │   ├── domelementtype@1.1.3
│ │ │   └── entities@1.1.1
│ │ ├── entities@1.0.0
│ │ └─┬ readable-stream@1.1.13
│ │   ├── core-util-is@1.0.1
│ │   ├── inherits@2.0.1
│ │   ├── isarray@0.0.1
│ │   └── string_decoder@0.10.31
│ ├── nwmatcher@1.3.6
│ ├── parse5@1.5.0
│ ├─┬ request@2.61.0
│ │ ├── aws-sign2@0.5.0
│ │ ├─┬ bl@1.0.0
│ │ │ └─┬ readable-stream@2.0.2
│ │ │   ├── core-util-is@1.0.1
│ │ │   ├── inherits@2.0.1
│ │ │   ├── isarray@0.0.1
│ │ │   ├── process-nextick-args@1.0.2
│ │ │   ├── string_decoder@0.10.31
│ │ │   └── util-deprecate@1.0.1
│ │ ├── caseless@0.11.0
│ │ ├─┬ combined-stream@1.0.5
│ │ │ └── delayed-stream@1.0.0
│ │ ├── extend@3.0.0
│ │ ├── forever-agent@0.6.1
│ │ ├─┬ form-data@1.0.0-rc3
│ │ │ └── async@1.4.2
│ │ ├─┬ har-validator@1.8.0
│ │ │ ├── bluebird@2.9.34
│ │ │ ├─┬ chalk@1.1.1
│ │ │ │ ├── ansi-styles@2.1.0
│ │ │ │ ├── escape-string-regexp@1.0.3
│ │ │ │ ├─┬ has-ansi@2.0.0
│ │ │ │ │ └── ansi-regex@2.0.0
│ │ │ │ ├─┬ strip-ansi@3.0.0
│ │ │ │ │ └── ansi-regex@2.0.0
│ │ │ │ └── supports-color@2.0.0
│ │ │ ├─┬ commander@2.8.1
│ │ │ │ └── graceful-readlink@1.0.1
│ │ │ └─┬ is-my-json-valid@2.12.2
│ │ │   ├── generate-function@2.0.0
│ │ │   ├─┬ generate-object-property@1.2.0
│ │ │   │ └── is-property@1.0.2
│ │ │   ├── jsonpointer@2.0.0
│ │ │   └── xtend@4.0.0
│ │ ├─┬ hawk@3.1.0
│ │ │ ├── boom@2.8.0
│ │ │ ├── cryptiles@2.0.4
│ │ │ ├── hoek@2.14.0
│ │ │ └── sntp@1.0.9
│ │ ├─┬ http-signature@0.11.0
│ │ │ ├── asn1@0.1.11
│ │ │ ├── assert-plus@0.1.5
│ │ │ └── ctype@0.5.3
│ │ ├── isstream@0.1.2
│ │ ├── json-stringify-safe@5.0.1
│ │ ├─┬ mime-types@2.1.5
│ │ │ └── mime-db@1.17.0
│ │ ├── node-uuid@1.4.3
│ │ ├── oauth-sign@0.8.0
│ │ ├── qs@4.0.0
│ │ ├── stringstream@0.0.4
│ │ ├── tough-cookie@2.0.0
│ │ └── tunnel-agent@0.4.1
│ └── xmlhttprequest@1.7.0
├── moment@1.6.2
├─┬ mysql@2.5.5
│ ├── bignumber.js@2.0.0
│ ├─┬ readable-stream@1.1.13
│ │ ├── core-util-is@1.0.1
│ │ ├── inherits@2.0.1
│ │ ├── isarray@0.0.1
│ │ └── string_decoder@0.10.31
│ └── require-all@1.0.0
├─┬ nodepie@0.5.5
│ ├─┬ fetch@0.3.6
│ │ └─┬ encoding@0.1.11
│ │   └── iconv-lite@0.4.11
│ ├─┬ iconv@2.1.10
│ │ └── nan@2.0.8
│ └─┬ xml2json@0.9.0
│   ├── hoek@2.14.0
│   ├─┬ joi@6.6.1
│   │ ├── isemail@1.1.1
│   │ ├── moment@2.10.6
│   │ └── topo@1.0.3
│   └─┬ node-expat@2.3.9
│     ├── bindings@1.2.1
│     ├─┬ debug@2.2.0
│     │ └── ms@0.7.1
│     └── nan@2.0.8
├── polyline-encoded@0.0.7
├─┬ scraper@0.0.9
│ └─┬ request@2.61.0
│   ├── aws-sign2@0.5.0
│   ├─┬ bl@1.0.0
│   │ └─┬ readable-stream@2.0.2
│   │   ├── core-util-is@1.0.1
│   │   ├── inherits@2.0.1
│   │   ├── isarray@0.0.1
│   │   ├── process-nextick-args@1.0.2
│   │   ├── string_decoder@0.10.31
│   │   └── util-deprecate@1.0.1
│   ├── caseless@0.11.0
│   ├─┬ combined-stream@1.0.5
│   │ └── delayed-stream@1.0.0
│   ├── extend@3.0.0
│   ├── forever-agent@0.6.1
│   ├─┬ form-data@1.0.0-rc3
│   │ └── async@1.4.2
│   ├─┬ har-validator@1.8.0
│   │ ├── bluebird@2.9.34
│   │ ├─┬ chalk@1.1.1
│   │ │ ├── ansi-styles@2.1.0
│   │ │ ├── escape-string-regexp@1.0.3
│   │ │ ├─┬ has-ansi@2.0.0
│   │ │ │ └── ansi-regex@2.0.0
│   │ │ ├─┬ strip-ansi@3.0.0
│   │ │ │ └── ansi-regex@2.0.0
│   │ │ └── supports-color@2.0.0
│   │ ├─┬ commander@2.8.1
│   │ │ └── graceful-readlink@1.0.1
│   │ └─┬ is-my-json-valid@2.12.2
│   │   ├── generate-function@2.0.0
│   │   ├─┬ generate-object-property@1.2.0
│   │   │ └── is-property@1.0.2
│   │   ├── jsonpointer@2.0.0
│   │   └── xtend@4.0.0
│   ├─┬ hawk@3.1.0
│   │ ├── boom@2.8.0
│   │ ├── cryptiles@2.0.4
│   │ ├── hoek@2.14.0
│   │ └── sntp@1.0.9
│   ├─┬ http-signature@0.11.0
│   │ ├── asn1@0.1.11
│   │ ├── assert-plus@0.1.5
│   │ └── ctype@0.5.3
│   ├── isstream@0.1.2
│   ├── json-stringify-safe@5.0.1
│   ├─┬ mime-types@2.1.5
│   │ └── mime-db@1.17.0
│   ├── node-uuid@1.4.3
│   ├── oauth-sign@0.8.0
│   ├── qs@4.0.0
│   ├── stringstream@0.0.4
│   ├── tough-cookie@2.0.0
│   └── tunnel-agent@0.4.1
├─┬ soupselect@0.2.0
│ └─┬ nodeunit@0.9.1
│   └─┬ tap@0.7.1
│     ├── buffer-equal@0.0.1
│     ├── deep-equal@1.0.1
│     ├─┬ difflet@0.2.6
│     │ ├── charm@0.1.2
│     │ ├── deep-is@0.1.3
│     │ └── traverse@0.6.6
│     ├─┬ glob@4.5.3
│     │ ├─┬ inflight@1.0.4
│     │ │ └── wrappy@1.0.1
│     │ ├─┬ minimatch@2.0.10
│     │ │ └─┬ brace-expansion@1.1.0
│     │ │   ├── balanced-match@0.2.0
│     │ │   └── concat-map@0.0.1
│     │ └─┬ once@1.3.2
│     │   └── wrappy@1.0.1
│     ├── inherits@2.0.1
│     ├─┬ mkdirp@0.5.1
│     │ └── minimist@0.0.8
│     ├─┬ nopt@3.0.3
│     │ └── abbrev@1.0.7
│     ├─┬ runforcover@0.0.2
│     │ └─┬ bunker@0.1.2
│     │   └─┬ burrito@0.2.12
│     │     ├── traverse@0.5.2
│     │     └── uglify-js@1.1.1
│     ├── slide@1.1.6
│     └── yamlish@0.0.7
├── underscore@1.3.3
├── underscore.string@2.3.3
├─┬ wolfram@0.2.7
│ ├─┬ libxmljs@0.8.1
│ │ └── bindings@1.0.0
│ └── request@2.1.1
└─┬ xml2js@0.2.8
  └── sax@0.5.8

hubot@8a0a47356723:/app# 
```

**Comment by arthurgeek** [comment]
I tried version `~3.2`, and the same error happened. Tried `3.1` and can't even connect:

```
[Wed Sep 16 2015 18:14:28 GMT+0000 (UTC)] DEBUG Loading adapter slack
[Wed Sep 16 2015 18:14:28 GMT+0000 (UTC)] INFO Connecting...
[Wed Sep 16 2015 18:14:34 GMT+0000 (UTC)] ERROR Received error failed to read response string
[Wed Sep 16 2015 18:14:34 GMT+0000 (UTC)] ERROR TypeError: Cannot call method 'close' of null
  at Client.reconnect (/app/node_modules/hubot-slack/node_modules/slack-client/src/client.coffee:159:5, <js>:180:15)
  at Client._onLogin (/app/node_modules/hubot-slack/node_modules/slack-client/src/client.coffee:55:32, <js>:78:25)
  at /app/node_modules/hubot-slack/node_modules/slack-client/src/client.coffee:1:1, <js>:3:61
  at IncomingMessage.<anonymous> (/app/node_modules/hubot-slack/node_modules/slack-client/src/client.coffee:522:13, <js>:614:22)
  at IncomingMessage.EventEmitter.emit (events.js:117:20)
  at _stream_readable.js:920:16
  at process._tickCallback (node.js:415:13)
```

**Comment by technicalpickles** [comment]
I know there's behavior in place already to [exit if there's an error from Slack](https://github.com/slackhq/hubot-slack/blob/master/src/slack.coffee#L49). You can see that here:

```
[Wed Sep 02 2015 17:32:53 GMT+0000 (UTC)] ERROR Received error "failed to read response string"
[Wed Sep 02 2015 17:32:53 GMT+0000 (UTC)] ERROR undefined
[Wed Sep 02 2015 17:32:53 GMT+0000 (UTC)] ERROR Exiting in 1 second
```

I think the bigger question is why you were getting that error in the first place. The second line, `undefined`, is trying to log the `Error`s, stack, so it seems like it's only a string that's coming back from slack-client. There's a few places around https://github.com/slackhq/node-slack-client/blob/master/src/client.coffee that `emit 'error'`, with things from the API, so it's probably from there. That code should be updated to create new `Error` so there is a stack at least.

Someone from Slack would need to comment about the "failed to read response string", if that's something that can come back from their API.

More generally, you'd want to use some sort of process management for your hubot, to make sure it starts back up from these kinds of errors.

**Comment by technicalpickles** [comment]
This actually looks really similar to https://github.com/slackhq/hubot-slack/issues/206

**Comment by wyattjoh** [comment]
Getting this issue when deployed:

```
hubot_1 | [Sat Nov 14 2015 02:53:15 GMT+0000 (UTC)] ERROR Last pong is too old: 10.004
hubot_1 | [Sat Nov 14 2015 02:53:15 GMT+0000 (UTC)] INFO Reconnecting in 1000ms
hubot_1 | [Sat Nov 14 2015 02:53:16 GMT+0000 (UTC)] INFO Attempting reconnect
hubot_1 | [Sat Nov 14 2015 02:53:16 GMT+0000 (UTC)] INFO Connecting...
hubot_1 | [Sat Nov 14 2015 02:53:45 GMT+0000 (UTC)] INFO Slack client closed, waiting for reconnect
hubot_1 | [Sat Nov 14 2015 02:54:37 GMT+0000 (UTC)] ERROR TypeError: Cannot read property 'code' of undefined
hubot_1 |   at SlackBot.error (/usr/src/app/node_modules/hubot-slack/src/slack.coffee:51:5, <js>:70:16)
hubot_1 |   at Client.<anonymous> (/usr/src/app/node_modules/hubot-slack/src/slack.coffee:1:1, <js>:3:61)
hubot_1 |   at emitOne (events.js:77:13)
hubot_1 |   at Client.emit (events.js:169:7)
hubot_1 |   at Client._onLogin (/usr/src/app/node_modules/slack-client/src/client.js:74:14)
hubot_1 |   at /usr/src/app/node_modules/slack-client/src/client.js:2:59
hubot_1 |   at ClientRequest.<anonymous> (/usr/src/app/node_modules/slack-client/src/client.js:684:18)
hubot_1 |   at emitOne (events.js:77:13)
hubot_1 |   at ClientRequest.emit (events.js:169:7)
hubot_1 |   at TLSSocket.socketErrorListener (_http_client.js:265:9)
hubot_1 |   at emitOne (events.js:77:13)
hubot_1 |   at TLSSocket.emit (events.js:169:7)
hubot_1 |   at emitErrorNT (net.js:1253:8)
hubot_1 |   at doNTCallback2 (node.js:450:9)
hubot_1 |   at process._tickCallback (node.js:364:17)
```

Which seems to be related to the return from the API: https://github.com/slackhq/hubot-slack/blob/1c49569b52b7774e07e158f4909b3a8e3a3a72c2/src/slack.coffee#L51

**Comment by FrenchBen** [comment]
:+1: on having the same issue - Looking forward to a fix

**Comment by aaronsky** [comment]
Getting the same error. 

```
[Tue Feb 09 2016 21:00:11 GMT-0500 (EST)] ERROR Last pong is too old: 13.958
[Tue Feb 09 2016 21:00:11 GMT-0500 (EST)] INFO Reconnecting in 1000ms
[Tue Feb 09 2016 21:00:11 GMT-0500 (EST)] INFO Slack client closed, waiting for reconnect
[Tue Feb 09 2016 21:00:12 GMT-0500 (EST)] INFO Attempting reconnect
[Tue Feb 09 2016 21:00:12 GMT-0500 (EST)] INFO Connecting...
[Tue Feb 09 2016 21:00:13 GMT-0500 (EST)] INFO Logged in as ibizan of extreme-serving, but not yet connected
[Tue Feb 09 2016 21:00:13 GMT-0500 (EST)] INFO Slack client now connected
[Tue Feb 09 2016 21:00:23 GMT-0500 (EST)] ERROR Last pong is too old: 10.009
[Tue Feb 09 2016 21:00:23 GMT-0500 (EST)] INFO Reconnecting in 1000ms
[Tue Feb 09 2016 21:00:23 GMT-0500 (EST)] INFO Slack client closed, waiting for reconnect
[Tue Feb 09 2016 21:00:24 GMT-0500 (EST)] INFO Attempting reconnect
[Tue Feb 09 2016 21:00:24 GMT-0500 (EST)] INFO Connecting...
[Tue Feb 09 2016 21:00:32 GMT-0500 (EST)] INFO Logged in as ibizan of extreme-serving, but not yet connected
[Tue Feb 09 2016 21:00:32 GMT-0500 (EST)] INFO Slack client now connected
[Tue Feb 09 2016 21:01:12 GMT-0500 (EST)] ERROR Last pong is too old: 11.65
[Tue Feb 09 2016 21:01:12 GMT-0500 (EST)] INFO Reconnecting in 1000ms
[Tue Feb 09 2016 21:01:12 GMT-0500 (EST)] INFO Slack client closed, waiting for reconnect
[Tue Feb 09 2016 21:01:13 GMT-0500 (EST)] INFO Attempting reconnect
[Tue Feb 09 2016 21:01:13 GMT-0500 (EST)] INFO Connecting...
[Tue Feb 09 2016 21:02:29 GMT-0500 (EST)] ERROR TypeError: Cannot read property 'code' of undefined
  at SlackBot.error (/Users/aaronsky/Repos/ibizan/node_modules/hubot-slack/src/slack.coffee:51:97)
  at Client.<anonymous> (/Users/aaronsky/Repos/ibizan/node_modules/hubot-slack/src/slack.coffee:1:1)
  at emitOne (events.js:77:13)
  at Client.emit (events.js:169:7)
  at Client._onLogin (/Users/aaronsky/Repos/ibizan/node_modules/hubot-slack/node_modules/slack-client/src/client.js:74:14)
  at /Users/aaronsky/Repos/ibizan/node_modules/hubot-slack/node_modules/slack-client/src/client.js:2:59
  at ClientRequest.<anonymous> (/Users/aaronsky/Repos/ibizan/node_modules/hubot-slack/node_modules/slack-client/src/client.js:684:18)
  at emitOne (events.js:77:13)
  at ClientRequest.emit (events.js:169:7)
  at TLSSocket.socketErrorListener (_http_client.js:259:9)
  at emitOne (events.js:77:13)
  at TLSSocket.emit (events.js:169:7)
  at emitErrorNT (net.js:1250:8)
  at doNTCallback2 (node.js:430:9)
  at process._tickCallback (node.js:344:17)
```

**Comment by FrenchBen** [comment]
This is also similar to #203

**Comment by DEGoodmanWilson** [comment]
There is…a chance…that this is resolved here https://github.com/slackhq/hubot-slack/pull/309
Willing to give it a try?

**Comment by DEGoodmanWilson** [comment]
Closing this as fixed in 4.0. Please reopen if I am mistaken!
